hedera-node/src/main/java/com/hedera/services/context/primitives/SignedStateViewFactory.java

```diff
@@ -0,0 +1,99 @@
+package com.hedera.services.context.primitives;
+
+import com.hedera.services.ServicesState;
+import com.hedera.services.context.ImmutableStateChildren;
+import com.hedera.services.context.MutableStateChildren;
+import com.hedera.services.state.migration.StateVersions;
+import com.swirlds.common.AutoCloseableWrapper;
+import com.swirlds.common.Platform;
+
+import java.time.Instant;
+
+/**
+ * Factory that provides view of last completed signed state and its state children.
```

Comment:
We actually want this to be a very "forgetful" factory. 😁

That is, _every time_ someone asks us for a new `StateChildren`, we should enter the `try-with-resources` block again, and try to use information from the most recent signed state available.

So we likely don't need a separate `hasValidSignedState()` method. Instead,
- In `tryToGet()`, we return `null` if there is no valid signed state; and a new `ImmutableStateChildren` if there is one.
- In `tryToUpdate()`, we update from latest valid and return `true` if possible; just return `false` otherwise.

Reply:
Removed setting `state` in constructor and called `getLatestSignedState()` method to get the latest state in `tryToGet()` and `tryToUpdate()` .
Returned `Optional.empty()` when state is invalid `tryToGet()` and modified others as suggested